Makefile

```diff
@@ -167,8 +168,8 @@ gen-test-golden: test-unit
 # dependencies to the go.mod file. To avoid that we change to a directory
 # without a go.mod file when downloading the following dependencies
 
-$(GOX):
-	(cd /; go install github.com/mitchellh/gox@v1.0.2-0.20220701044238-9f712387e2d2)
+$(GORELEASER):
+	(cd /; GOTOOLCHAIN=$(GOTOOLCHAIN) go install github.com/goreleaser/goreleaser@latest)
```

Comment:
is there is a specific reason why we want/need to lock the goreleaser install to use the same go toolchain as in go.mod?

Reply:
The only reason is it automates upgrading without getting too far out.

Reply:
Except, we don't built based on the version in the `go.mod` file. If we did that, we would need to update it for every Go release that comes out. We don't track every time a Go release comes out. For example, it is currently set for `1.24.0`. That means the standard library used would be that with 1.24.0 which includes known CVEs.

For builds we use the major/minor specified in `.github/env`. The system always does the latest patch release. Can you please remove this GOTOOLCHAIN specification.